### ISSUE TITLE
User Information Submission System

### ISSUE THREAD

**Issue by makeworld-the-better-one** [tomeshnet/toronto-community-network#83]
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Task Summary

🎟️ **Re-ticketed from:** #
📅 **Due date:** September 29, 2020
🎯 **Success criteria:** A system exists to provide us with the necessary contact and location info to record a proposed node.

## Needed Info
- Contact email
- Optional other contact info (Matrix, phone, Signal, etc)
- Latitude, Longitude
	- Submitting address should be allowed too
- ~~Altitude or~~ building floor (we'll convert)
- First/Full name
- Are you okay with this info being made public (anonymously)?

## Optional Info
- Optional other contact info (Matrix, phone, Signal, etc)
- Questions
- Personal contact within tomesh if exists
- How did you find out about tomesh

## To Do

- [x] Decide on system - Google Form, web form with formspree, etc  EDIT: We're using next cloud forms.
- [x] Create nextcloud form
- [x] Update website homepage to link to the form and replace current "get involved" section

**Comment by darkdrgn2k** [comment]
> An email and non-email system exists  

I would limit the system to email only. However  I would reword the criteria to simply:
`A system exists to provide us with the necessary contact and location info to record a proposed node.`

**Comment by darkdrgn2k** [comment]
My suggestions would be email only for now
- We get contact information 
- Low bar to entry (have an email address)

Then transpose the data to a next cloud sheet


i would make it simple and open ended.

Who are you
Where do you live
What are your expectations/what do you want.

We have seen in the past matrix and other mechanisms fail over time.  If we need to get a hold of the person (ie if there is a problem with the node) i would like to be able to contact the person and not hope they are still on matrix.

**Comment by makeworld-the-better-one** [comment]
> We have seen in the past matrix and other mechanisms fail over time. If we need to get a hold of the person (ie if there is a problem with the node) i would like to be able to contact the person and not hope they are still on matrix.

That's why contact email was required, and additional contact info is optional. I thought it would be good to provide another way, as a backup in case that person doesn't check email or moves addresses, etc.

**Comment by darkdrgn2k** [comment]
> > We have seen in the past matrix and other mechanisms fail over time. If we need to get a hold of the person (ie if there is a problem with the node) i would like to be able to contact the person and not hope they are still on matrix.
> 
> That's why contact email was required, and additional contact info is optional. I thought it would be good to provide another way, as a backup in case that person doesn't check email or moves addresses, etc.

Fair, but i would also caution that there will be a main communication point, so we dont have to draft communication for all the different media of communication.

**Comment by darkdrgn2k** [comment]
Looking at People's open onboarding process for nodes

Some one in their channel was asking about deploying, the response was:

> nice! Would you mind firing off a quick email to info@peoplesopen.net so we have your contact info and address? We have a few planned node installs. Nearest one is close to Adams Point

Asked for clarification and their internal process.

**Comment by infectedroot** [comment]
NextCloud does forms. We're already using NC to store information about other things Toronto Mesh.

You can make forms that are anon accessible and gather information.

Check out this demo form (when logged into NC) for things to edit (and view responses):
https://cloud.tomesh.net/apps/forms/EfmCbb7H9WFHZ49Q/edit

And while not logged in...
https://cloud.tomesh.net/apps/forms/EfmCbb7H9WFHZ49Q

I'm also +1 on the email method.

**Comment by infectedroot** [comment]
@benhylau The form for the website is available at https://cloud.tomesh.net/apps/forms/3eJKtBKQ3m4Br27B

Let me know if you guys want to see any more changes.

**Comment by benhylau** [comment]
thanks that's now added to https://link.tomesh.net/join (once the PR merges anyways)

**Comment by benhylau** [comment]
@infectedroot please update the form with the following questions and make sure the permission settings allow unauthenticated users to submit the form.

```
Please provide us with your information if you'd like to contribute to Toronto Mesh, or to join our network! Someone in our Project Operations team will be in contact with you shortly.

If you'd like to contribute to our working groups, check out https://link.tomesh.net/wg for more information.

First name

Last name

Contact email

Other contact information (e.g. Signal, Matrix)

How would you like to get involved? For example, is there a particular working group you are interested in joining?

Do you have any questions for us?

How did you find out about Toronto Mesh?

Do you already know someone at Toronto Mesh? If so, let us know who they are!

Would you like to maintain a Toronto Community Network node? If not, please skip the rest of the questions and submit this form.

Node address

Node latitude

Node longitude

Node elevation

Node direction

Would you like us to make your prospective node information public anonymously? This will help us map out prospective locations on our map https://tomesh.net/map/.
```

**Comment by darkdrgn2k** [comment]
Discussions at a meeting we had was  NOT to use a form and just use an email address instead.

Some points as i remember them

- There is some design thinking planning going on about on boarding.  This may yield a much better and integrated workflow
- Using email as a means of contact means
   - We have confirmation of the contact information of the person (because they emailed us)
   - Replies to email are more likely to pass spam filters
   - It has very low overhead to implement and we don't have cycles to spend on this kind of stuff right now
   - Sending an email requires a little bit of effort. This would act as a filter to eliminate individuals who do not want to participate.
   - A more free form approach would help us understand how people understand data  IE Longitude and Latitude may be hard for an individual to come up with, or even elevation.

**Comment by darkdrgn2k** [comment]
> Would you like us to make your prospective node information public anonymously?

Why is this an option?
Are we going to now maintain separate lists?
Who gets access to the non public information?

`Would you like to maintain a Toronto Community Network node` 

What does that even mean?

**Comment by benhylau** [comment]
>Would you like us to make your prospective node information public anonymously?

A trusted spreadsheet I imagine? Either way we need to hide their email right?

>Would you like to maintain a Toronto Community Network node? If not, please skip the rest of the questions and submit this form.

It's meant to ask whether someone wants to have a node at home. How would you phrase it? We can change it to what you think is more appropriate?

cc. @infectedroot

**Comment by infectedroot** [comment]
If we're having so many issues sharing a simple form administratively within our NextCloud instance, then that means that my account will be the only account to see the responses as well. This causes an issue for any responses to the form, as I'll be the only point of contact for it. This isn't the wisest thing for us.

As @darkdrgn2k  suggested, it's actually really easy for us to get the information via email. Maybte it's time to go "back to basics" and use email.

Let's just use a this fancy [link as the call the action](mailto:operations@tomesh.net?subject=I'd%20like%20to%20join%20the%20Toronto%20Mesh!&body=Please%20provide%20us%20with%20your%20information%20if%20you'd%20like%20to%20contribute%20to%20Toronto%20Mesh%2C%20or%20to%20join%20our%20network!%20Someone%20in%20our%20Project%20Operations%20team%20will%20be%20in%20contact%20with%20you%20shortly.%0D%0A%0D%0AIf%20you'd%20like%20to%20contribute%20to%20our%20working%20groups%2C%20check%20out%20https%3A%2F%2Flink.tomesh.net%2Fwg%20for%20more%20information.%0D%0A%0D%0AFirst%20name%3A%0D%0ALast%20name%3A%0D%0AContact%20email%3A%0D%0AOther%20contact%20information%20(e.g.%20Signal%2C%20Matrix)%3A%0D%0A%0D%0AHow%20would%20you%20like%20to%20get%20involved%3F%20For%20example%2C%20is%20there%20a%20particular%20working%20group%20you%20are%20interested%20in%20joining%3F%0D%0A%0D%0ADo%20you%20have%20any%20questions%20for%20us%3F%0D%0A%0D%0AHow%20did%20you%20find%20out%20about%20Toronto%20Mesh%3F%0D%0A%0D%0ADo%20you%20already%20know%20someone%20at%20Toronto%20Mesh%3F%20If%20so%2C%20let%20us%20know%20who%20they%20are!%0D%0A%0D%0AWould%20you%20like%20to%20maintain%20a%20Toronto%20Community%20Network%20node%3F%20Let%20us%20know%20and%20we'll%20get%20back%20to%20you%20with%20more%20information.%0D%0A%0D%0A). I built it with [mailtolink.me](https://mailtolink.me/). I omitted some of the node instructions due to length issues - and it complicates the action. Let's just get their information to keep in touch and go from there.

When an onboarding request comes in, we can cross-reference it against the proper contrib/members spreadsheet to check if they're onboarded.

cc. @benhylau

**Comment by benhylau** [comment]
I think that's perfect and agree on omitting the node info. We can do that by responding to them to guide them through the questionnaire, or invite them to a meeting so we can put the location on the map together. Thanks for setting this up!

**Comment by makeworld-the-better-one** [comment]
I think it's important to have the email address and questions available on a website as well, whatever the method is. For example, I don't have a default email client setup, and so if I didn't know how to read mailto links (or that they even existed), I wouldn't really be able to participate with this method.

**Comment by darkdrgn2k** [comment]
I don't want to derail the progress here. Just very frustrating that time is being spent debating things already debated and not because new issues arose.

> I think it's important to have the email address and questions available on a website as well, whatever the method is. For example, I don't have a default email client setup, and so if I didn't know how to read mailto links (or that they even existed), I wouldn't really be able to participate with this method.

As noted In the initial conversation where "i don't have a mail client" first came up,  i pointed out that although sending an email requires a small amount of additional work. This would act as a filter to separate the people who do not with to take on this extra work, and therefore would not be willing to put work into the project themselves. 

I will say that an email method also invites spam, so might want to do a throw away email to use if it becomes to great. 

Its not an imperfect solution and i did note that work is being done for onboarding AND that we would not be getting a high volume of requests initially (possibly ever)


Location is a VERY important field and i don't think we should remove it. Without it we cannot plan or rally support for community deployment so it needs to be asked.  Until this can be done from the map the question should be open ended as much as it can be.  

Lat/Lon/Altitude is useful but Address/Floor would work as well.  This is why an open question would probably work better then.  Again clarifying questions could be asked.

Gets complicated when using for example altitude its also important to note what this altitude is, Sea level, Ground Level etc.

I know quite a few people who may be interested in doing a node but probably cant/wont make a meeting. Especially when it returns to physical.

So perhaps just a "where are you located" or "where is the node to be located" is open ended enough to get a sense of where they are coming from.

**Comment by darkdrgn2k** [comment]
Don't know if this was covered at the mast meeting, but I think you may also want to separate Contributor on boarding and Node request on boarding.

This may discourage people to state their interest if wg level involvement is required.  

Perhaps either separate the call two actions into two funnels, or split the funnel after the call to action.

NOTE: we already have a email for onboarding, we could reuse that in the above suggested method too.
To @makeworld-the-better-one point we should also have a manual explanation of how to do it if mailto does not work.

**Comment by Shrinks99** [comment]
Alright, looks like this has to be revisited!  Steps to completion:

- [ ] Replace `get involved` page text with our new request for people to email us
  - Get rid of funding section
  - Keep chat section
- [ ] Replace `link.tomesh.net/join` with a link to our new get involved page
- [ ] Update homepage to link to `get involved`

**Comment by infectedroot** [comment]
@Shrinks99 [The new mailto is here](mailto:hello@tomesh.net?subject=I'd%20like%20to%20join%20the%20Toronto%20Mesh!&body=Please%20provide%20us%20with%20your%20information%20if%20you'd%20like%20to%20join%20Toronto%20Mesh!%20Someone%20in%20our%20Project%20Operations%20team%20will%20be%20in%20contact%20with%20you%20shortly.%0D%0A%0D%0AIf%20you'd%20like%20to%20contribute%20to%20our%20working%20groups%2C%20check%20out%20https%3A%2F%2Flink.tomesh.net%2Fwg%20for%20more%20information.%0D%0A%0D%0AHow%20would%20you%20like%20to%20get%20involved%3F%20For%20example%2C%20is%20there%20a%20particular%20working%20group%20you%20are%20interested%20in%20joining%3F%0D%0A%0D%0A...%0D%0A%0D%0AHow%20did%20you%20find%20out%20about%20Toronto%20Mesh%3F%0D%0A%0D%0A...%0D%0A%0D%0ADo%20you%20already%20know%20someone%20at%20Toronto%20Mesh%3F%20If%20so%2C%20let%20us%20know%20who%20they%20are!%0D%0A%0D%0A...%0D%0A%0D%0AWould%20you%20like%20to%20maintain%20a%20Toronto%20Community%20Network%20node%3F%20Let%20us%20know%20and%20we'll%20get%20back%20to%20you%20with%20more%20information.%0D%0A%0D%0A...%0D%0A%0D%0ADo%20you%20have%20any%20other%20questions%20for%20us%3F%0D%0A%0D%0A...%0D%0A%0D%0APreferred%20name%3A%0D%0AOther%20contact%20information%20(e.g.%20Signal%2C%20Matrix)%3A)

**Comment by Shrinks99** [comment]
Closing, all other discussion should happen [here](https://github.com/tomeshnet/tomesh.net/pull/299).  When we revisit our on-boarding system next we will make a new issue ;P
